src/views/address/search.njk

```diff
@@ -47,4 +47,26 @@
     dynamic: false
   }) }}
 
+  <script nonce="{{ cspNonce }}">
+    var formSubmitted = false;
+    submitSpinner()
+    function delayDisableButton(button) {
+      button.className += ' button--spinner'
+      setTimeout(function() {
+            button.setAttribute('disabled', true)
+        }, 200);
+    }
+    function submitSpinner() {
+      var select = document.getElementById('continue')
+      select.addEventListener('click', function (event) {
+        if (formSubmitted) {
+          event.preventDefault()
+        } else {
+          formSubmitted = true;
+          delayDisableButton(event.target);
+        }
+      })
+    }
+  </script>
```

Comment:
I know you've probably copied this from somewhere, is this how it's repeated for every page. we should define this once and reuse on pages that need the functionality.

Reply:
@jkunle I completely agree.

I think this is something that is on the FEC's teams backlog, although I don't know what the priority is.

Reply:
Also the on each page the functions are exposed to the global. it would probably be better to wrap them in an IIF
```
(function() {
.....your code goes here
})()
```
maybe externalize this so you can do something like `<script src="path/to/spinner.js" nonce="{{ cspNonce }}"></script>`

Reply:
I've never heard this @jkunle would you be able to elaborate on why it's better to wrap in an IIF?

Reply:
Agree that would be a better approach, I just copied KBV and Nino to keep it consistent. 

I've moved the `script` code so it is now at least reusable within this repo. I'd think any further than that is possibly outside the scope of this ticket, possibly worth a new tech debt ticket?

Reply:
@Caitlin-cooling IIF are a classic ES5 pattern to avoid putting things in the global scope which is what happens when just write pure javascript pre ES6 https://developer.mozilla.org/en-US/docs/Glossary/IIFE

Reply:
@mikebeeby I think with what you've done this has reached the good enough threshold for me. There is just one place now to change. Given we don't use any other javascript libraries, the chances of collision on the global scope is very very small. However is generally a good practice

Reply:
```
  (function() {
    var formSubmitted = false;
    
    function delayDisableButton(button) {
      button.className += ' button--spinner';
      setTimeout(function() {
        button.setAttribute('disabled', true);
      }, 200);
    }
    
    function submitSpinner() {
      var select = document.getElementById('continue');
      select.addEventListener('click', function (event) {
        if (formSubmitted) {
          event.preventDefault();
        } else {
          formSubmitted = true;
          delayDisableButton(event.target);
        }
      });
    }
    
    submitSpinner();
  })();
```